src/v/archival/archival_policy.h

```diff
@@ -44,6 +61,21 @@ struct upload_candidate_with_locks {
     std::vector<ss::rwlock::holder> read_locks;
 };
 
+/// Wraps an error with an offset range, so that no
+/// further upload candidates are created from this offset range.
+struct skip_offset_range {
+    model::offset begin;
```

Comment:
begin_inclusive/end_inclusive? to be consistent with the naming in upload candidate struct

Reply:
is begin even used anywhere? shall we remove it?

Reply:
changed, I think `begin_offset` is somewhat useful when looking at logs to see what range is skipped, so I kept it as a field. Initially I also thought it would be useful in unit tests to assert that the correct range is skipped but I see there are no tests checking the begin offset.